### PR TITLE
Clean up NPM dependencies and update node to v20

### LINKED ISSUE #274: Clean up NPM dependencies and update node to v20
### Is your enhancement related to a problem? Please describe.

This is to align with the current tooling versions used by WordPress - https://make.wordpress.org/core/2023/12/20/updating-wordpress-to-use-more-modern-versions-of-node-js-npm-2/

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

This PR cleans up NPM dependencies and updates the node version to `20.11.0`
Closes #274

### Changelog Entry

> Changed - Clean up NPM dependencies and update node to v20

### Credits

Props @Sidsector9 

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: package-lock.json
Comment (peterwilsoncc) [inline package-lock.json:1]: The lock file contains several merge conflict artifacts.

File: webpack.config.js [hunk 1]
```diff
@@ -55,4 +55,7 @@ module.exports = {
 			],
 		}),
 	],
+	externals: {
+		'@wordpress/plugins': 'wp.plugins',
+	}
```
Comment (peterwilsoncc) [inline webpack.config.js]: This isn't required as it's handled by the `@wordpress/dependency-extraction-webpack-plugin` in wp-scripts.

**Review by peterwilsoncc — CHANGES_REQUESTED** [review]
The E2E Build workflow is logging a node 16 deprecation error, it would be good to update while working on node fixes.

The E2E tests are failing in the WP Minimum tests. The same is happening on trunk so I suspect this expected due to a class change on the snackbar component. It's looking for `.components-snackbar`.

Other notes:

* `npm run build` works successfully
* wp-env: starts, stops and destroy work as expected
* cypress: runs locally and in e2e workflows

I'm on OS this week so can have a look at the merge conflicts and workflows.

**Comment by peterwilsoncc** [comment]
@jeffpaul I've bumped a bunch of actions to avoid Node deprecation warnings. In some cases I couldn't as they are using other repos that either define an out of date node version.

It looks like the upload-artifact/download-artifact versions will need to be updated in sync across all our repos as v3 uploads are incompatible with v4 downloads.

**Comment by jeffpaul** [comment]
@peterwilsoncc is this PR otherwise approved to merge?

**Review by peterwilsoncc — APPROVED** [review]
@jeffpaul Yep, otherwise looks good.

**Pull request merged**